appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/model/parameters/RequestBodyImpl.java

```diff
@@ -55,19 +57,24 @@
 
     private String description;
     private Content content = new ContentImpl();
-    private List<ContentImpl> contents = new ArrayList<>();
     private Boolean required;
     private String ref;
 
-    public static RequestBody createInstance(AnnotationModel annotation, ApiContext context) {
+    public static RequestBodyImpl createInstance(AnnotationModel annotation, ApiContext context) {
         RequestBodyImpl from = new RequestBodyImpl();
         from.setDescription(annotation.getValue("description", String.class));
         from.setRequired(annotation.getValue("required", Boolean.class));
         String ref = annotation.getValue("ref", String.class);
         if (ref != null && !ref.isEmpty()) {
             from.setRef(ref);
         }
-        extractAnnotations(annotation, context, "content", ContentImpl::createInstance, from.getContents());
+
+        final List<ContentImpl> contents = createList();
+        extractAnnotations(annotation, context, "content", ContentImpl::createInstance, contents::add);
+        for (ContentImpl content : contents) {
+            ContentImpl.merge(content, from.content, true, context);
```

Comment:
As per the existing design of OpenAPI Impl, `merge` operations are invoked after the creation of the instances.
````
    public void visitOpenAPI(AnnotationModel definition, AnnotatedElement element, ApiContext context) {
        OpenAPIImpl.merge(OpenAPIImpl.createInstance(definition, context), context.getApi(), true, context);
    }
````

Reply:
What do you mean?

Reply:
Discussed over impl design in the DM. PR LGTM.